pandas/io/orc.py

```diff
@@ -55,3 +56,74 @@ def read_orc(
     with get_handle(path, "rb", is_text=False) as handles:
         orc_file = orc.ORCFile(handles.handle)
         return orc_file.read(columns=columns, **kwargs).to_pandas()
+
+
+def to_orc(
+    df: DataFrame,
+    path: FilePathOrBuffer = None,
+    engine: str = 'pyarrow',
+    index: bool = None,
+    **kwargs
+) -> bytes:
+    """
+    Write a DataFrame to the ORC format.
+    Parameters
+    ----------
+    df : DataFrame
+    path : str or file-like object, default None
+        If a string, it will be used as Root Directory path
+        when writing a partitioned dataset. By file-like object,
+        we refer to objects with a write() method, such as a file handle
+        (e.g. via builtin open function). If path is None,
+        a bytes object is returned. Note that currently the pyarrow
+        engine doesn't work with io.BytesIO.
+    engine : {{'pyarrow'}}, default 'pyarrow'
+        Parquet library to use, or library it self, checked with 'pyarrow' name
+        and version >= 5.0.0
+    index : bool, default None
+        If ``True``, include the dataframe's index(es) in the file output. If
+        ``False``, they will not be written to the file.
+        If ``None``, similar to ``infer`` the dataframe's index(es)
+        will be saved. However, instead of being saved as values,
+        the RangeIndex will be stored as a range in the metadata so it
+        doesn't require much space and is faster. Other indexes will
+        be included as columns in the file output.
+    kwargs
+        Additional keyword arguments passed to the engine
+    Returns
+    -------
+    bytes if no path argument is provided else None
+    """
+    if index is None:
+        index = df.index.names[0] is not None
+
+    if isinstance(engine, str):
+        engine = import_optional_dependency(engine, min_version='5.0.0')
+    else:
```

Comment:
`engine` is annotated to be of type string. Could use `types.ModuleType` but will need mypy ignores?

Reply:
Thanks! Fixed!

Reply:
Sorry, I meant for the type of engine, not in the if condition. Keeping the `else` should be fine (and probably also better, in case a user provides something that isn't a string nor a module).

Reply:
Thanks! I removed `types.ModuleType` as well as code that allows an `engine` of such a type.